### ISSUE TITLE
Malformed TiffFile raises ValueError

### ISSUE THREAD

**Issue by MrShark** [python-pillow/Pillow#1544]
While fuzzing imagesfiles I found a file that causes `Image.open(infile).load()` to give the following stack trace:

```
Traceback (most recent call last):
  File "./test2.py", line 8, in <module>
    Image.open(infile).load()
  File "/install/local/lib/python2.7/site-packages/PIL/Image.py", line 2284, in open
    im = _open_core(fp, filename, prefix)
  File "/install/local/lib/python2.7/site-packages/PIL/Image.py", line 2274, in _open_core
    im = factory(fp, filename)
  File "/install/local/lib/python2.7/site-packages/PIL/ImageFile.py", line 100, in __init__
    self._open()
  File "/install/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 772, in _open
    self._seek(0)
  File "/install/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 833, in _seek
    self._setup()
  File "/install/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 1021, in _setup
    h = self.tag_v2.get(ROWSPERSTRIP, ysize)
  File "/install/lib/python2.7/_abcoll.py", line 382, in get
    return self[key]
  File "/install/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 352, in __getitem__
    self[tag] = handler(self, data, self.legacy_api)  # check type
  File "/install/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 366, in __setitem__
    self._setitem(tag, value, self.legacy_api)
  File "/install/local/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 407, in _setitem
    dest[tag], = values
ValueError: too many values to unpack
```

The image file is uploaded here: 
[tiff_image.txt](https://github.com/python-pillow/Pillow/files/36738/tiff_image.txt)

**Comment by radarhere** [comment]
For the next person who looks at this, the `values` variable at that point contains `(37396, 52970, 35421, 30692, 24655, 24841, 46368, 26699, 14119)`

**Comment by wiredfool** [comment]
I think that this is a case of throwing a different error than usual on a malformed file. I don't think it leads to any interesting behavior. We should catch that error at a reasonably high level, perhaps in Image.open._opencore and return the usual error for a malformed file. 

As an aside, I'd be interested in getting afl to fuzz tiffs that follow the libtiff pathway because that's likely to lead to interesting C level bugs, rather than different python errors. The easiest way to trigger that is specify a compression in the atts, or to force the LIBTIFF flags in TiffImagePlugin.

**Comment by radarhere** [comment]
Just to clearly get permission, are you happy for the image to be distributed as a test image under Pillow's license?

**Comment by wiredfool** [comment]
* [ ] More careful handling and validation of required tiff tags

**Comment by radarhere** [comment]
This now throws a warning instead of an error. Testing, this was addressed by #2719
